connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/DefaultInboundConnectorContextFactory.java

```diff
@@ -0,0 +1,91 @@
+import io.camunda.connector.api.validation.ValidationProvider;
+import io.camunda.connector.feel.FeelEngineWrapper;
+import io.camunda.connector.runtime.core.inbound.correlation.InboundCorrelationHandler;
+import io.camunda.connector.runtime.core.secret.SecretProviderAggregator;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.function.Consumer;
+
+public class DefaultInboundConnectorContextFactory implements InboundConnectorContextFactory {
+  private final ObjectMapper objectMapper;
+  private final InboundCorrelationHandler correlationHandler;
+  private final SecretProviderAggregator secretProviderAggregator;
+  private final ValidationProvider validationProvider;
+  private final OperateClientAdapter operateClientAdapter;
+  private final FeelEngineWrapper feelEngineWrapper;
+
+  public DefaultInboundConnectorContextFactory(
+      final ObjectMapper mapper,
+      final InboundCorrelationHandler correlationHandler,
+      final SecretProviderAggregator secretProviderAggregator,
+      final ValidationProvider validationProvider,
+      final OperateClientAdapter operateClientAdapter,
+      final FeelEngineWrapper feelEngineWrapper) {
+    this.objectMapper = mapper;
+    this.correlationHandler = correlationHandler;
+    this.secretProviderAggregator = secretProviderAggregator;
+    this.validationProvider = validationProvider;
+    this.operateClientAdapter = operateClientAdapter;
+    this.feelEngineWrapper = feelEngineWrapper;
+  }
+
+  @Override
+  public <T extends InboundConnectorExecutable<?>> InboundConnectorContext createContext(
+      final InboundConnectorDefinitionImpl newConnector,
+      final Consumer<Throwable> cancellationCallback,
+      final Class<T> executableClass) {
+
+    InboundConnectorContext inboundContext =
+        new InboundConnectorContextImpl(
+            secretProviderAggregator,
+            validationProvider,
+            newConnector,
+            correlationHandler,
+            cancellationCallback,
+            objectMapper);
+
+    // Inspects the generic type of InboundConnectorExecutable to determine the type of context
+    // needed and initializes it accordingly.
+    Type[] genericInterfaces = executableClass.getGenericInterfaces();
+    for (Type genericInterface : genericInterfaces) {
+      if (genericInterface instanceof ParameterizedType parameterizedType) {
+        Type[] actualTypeArguments = parameterizedType.getActualTypeArguments();
+        if (actualTypeArguments.length > 0
+            && actualTypeArguments[0].equals(InboundIntermediateConnectorContext.class)) {
+          inboundContext =
+              new InboundIntermediateConnectorContextImpl(
+                  inboundContext,
+                  operateClientAdapter,
+                  validationProvider,
+                  feelEngineWrapper,
+                  objectMapper);
+          break;
+        }
+      }
+    }
```

Comment:
Please extract into separate method and have a conditional return statement: isIntermediateContext()

Dont create the other context if not needed.